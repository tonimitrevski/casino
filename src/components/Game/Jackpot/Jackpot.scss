@import "../../../style/variables";

.jackpot {
  position: absolute;
  width: 100%;
  display: flex;
  justify-content: center;
  top: 15px;
  left: 0;
  color: $color-ribon-label-buttons;
  padding: 3px 8px;
  &__container {
    display: flex;
    justify-content: center;
    background-color: rgba(0, 0, 0, 0.3);
    width: 100%;
    span {
      font-weight: bold;
    }
  }
}
