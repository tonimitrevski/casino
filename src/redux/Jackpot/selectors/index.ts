import {StoreType} from "../../store.type";

export const getJackpotsSelector = (state: StoreType) => state.jackpots;
