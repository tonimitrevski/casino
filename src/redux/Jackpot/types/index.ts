import {FetchJackPotAction} from "./fetch-jackpot-action.type";

export type JackpotActionTypes = FetchJackPotAction;
