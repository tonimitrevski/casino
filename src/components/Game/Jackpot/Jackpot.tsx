import "./Jackpot.scss";
import React from "react";

export const Jackpot = (props : { amount: number }) => {
  const { amount } = props;
  return (
    <div className="jackpot">
      <div className="jackpot__container">
        <span className="amount">{currencyFormat(amount)}</span>
      </div>
    </div>
  );
}

function currencyFormat(amount: number) {
  return new Intl.NumberFormat('en-UK', {
    style: 'currency',
    currency: 'GBP',
    minimumFractionDigits: 2
  }).format(amount)
}

export default Jackpot;
