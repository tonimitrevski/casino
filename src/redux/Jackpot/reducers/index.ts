import {JackpotActionTypes} from "../types";
import {FetchJackpotAction} from "../types/fetch-jackpot-action.type";
import {JackpotsAggregate} from "../../../core/aggregate/jackpots.aggregate";
import {JackpotStateInterface} from "../types/jackpot-state.interface";
export const jackpotInitialState: JackpotStateInterface = { data: new JackpotsAggregate(), error: null};

const jackPotReducer = (state= jackpotInitialState, action: JackpotActionTypes) => {
  switch (action.type) {
    case FetchJackpotAction: {
      return {
        ...state,
        data: action.payload
      };
    }
    default: {
      return state;
    }
  }
};

export default jackPotReducer;
