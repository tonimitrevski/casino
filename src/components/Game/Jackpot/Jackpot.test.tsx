import React from 'react';
import {shallow} from "enzyme";
import Jackpot from "./Jackpot";

describe("Jackpot Component", () => {
  test('Test Jackpot component', () => {
    const wrapper = shallow(<Jackpot amount={1000} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.jackpot').length).toBe(1);
    expect(wrapper.find('.jackpot__container').length).toBe(1);
    expect(wrapper.find('.amount').length).toBe(1);
    expect(wrapper.find('.amount').text()).toBe("Â£1,000.00");
  });
});
