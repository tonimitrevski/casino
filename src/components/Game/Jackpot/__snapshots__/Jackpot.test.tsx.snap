// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jackpot Component Test Jackpot component 1`] = `
<div
  className="jackpot"
>
  <div
    className="jackpot__container"
  >
    <span
      className="amount"
    >
      Â£1,000.00
    </span>
  </div>
</div>
`;
